backend/src/db/repositories/CommentRepository.ts

```diff
@@ -43,23 +43,27 @@ export default class CommentRepository {
             });
     }
 
-    async getUserComments(userId: number, forUserId: number, page: number, perPage: number): Promise<CommentRawWithUserData[]> {
+    async getUserComments(userId: number, forUserId: number, filter: string, page: number, perPage: number): Promise<CommentRawWithUserData[]> {
         const limitFrom = (page - 1) * perPage;
-        return await this.db.query(`
+        const query = `
                 select c.*, v.vote
                     from comments c
                       left join comment_votes v on (v.comment_id = c.comment_id and v.voter_id = :for_user_id)
                     where
-                      c.author_id = :user_id and c.deleted = 0
+                      c.author_id = :user_id and c.deleted = 0 ${filter ? ' and c.source like :filter ' : ''}
                     order by c.created_at desc
                     limit :limit_from, :limit_count
-            `,
-            {
-                user_id: userId,
-                for_user_id: forUserId,
-                limit_from: limitFrom,
-                limit_count: perPage
-            });
+            `;
+          const params = {
+              user_id: userId,
+              for_user_id: forUserId,
+              limit_from: limitFrom,
+              limit_count: perPage
+          };
+        if (filter) {
+            params['filter'] = '%' + filter + '%';
```

Comment:
We should test this for potential abuse. E.g. super long queries with `%` in them.

Reply:
Fixed.